### DISCUSSION TITLE
Generated names usage

### DISCUSSION THREAD

**Discussion by suhtai** [kopykat-kt/kopykat#67] [Ideas]
Currently, we are generating names that are valid when Unicode escaped.

![image](https://user-images.githubusercontent.com/260697/195999626-39282779-f54e-4591-bc8b-de4a0c01b3c6.png)

However, they could be a bit awkward to use if a user of KopyKat wants to use the generated type. For instance, someone may create an extension function on the Mutable type.

So, in here, it may make more sense to use the name `MutablePerson` instead.

**Comment by serras** [comment]
I actually changed it to make it clear they were generated names, and in principle not something one would use directly.

Apart from that, simply prepending `Mutable` may conflict with another class, which is less possible when using `$Mutable`.

**Reply by suhtai** [reply]
Yeah, that makes sense. Although, we could make the constructor internal. After all, we have `toMutable` public 😅 

I thought about the clash before, yeah that's a potential issue. Eventually, I want to be able to add options per class so that the user can change each class prefixes, postfixes and the sort 🙂 Very low priority for v1.0 though

**Comment by serras** [comment]
In any case, I’m OK with changing the names to more readable ones if we expect people to use those classes.
